### PR TITLE
En 6314/no rating change in first epoch after hardfork

### PR DESCRIPTION
**Pull request by sasurobert**

### CODE REVIEW

File: process/peer/process.go [hunk 1]
```diff
@@ -954,21 +1004,30 @@ func (vs *validatorStatistics) display(validatorKey string) {
...
 	}
+
 	shardValidators := validators[shardID]
 	validatorsCount := len(shardValidators)
 	percentageRoundMissedFromTotalValidators := float64(missedRounds) / float64(validatorsCount)
 	leaderAppearances := uint32(percentageRoundMissedFromTotalValidators + 1 - math.SmallestNonzeroFloat64)
 	consensusGroupAppearances := uint32(float64(consensusGroupSize)*percentageRoundMissedFromTotalValidators +
 		1 - math.SmallestNonzeroFloat64)
 	ratingDifference := uint32(0)
+
```
Comment (iulianpascalau) [inline process/peer/process.go:1030]: So basically, won't decrease the rating if a consensus round is missed but we allow to increase the ratings on all peers (during the epoch grace period).

Comment (sasurobert) [reply process/peer/process.go:1030]: those who are working well, should be rewarded as well.

File: process/peer/process.go [hunk 2]
```diff
@@ -827,7 +874,10 @@ func (vs *validatorStatistics) updateValidatorInfo(validatorList []sharding.Vali
 			newRating = vs.rater.ComputeIncreaseValidator(shardId, peerAcc.GetTempRating())
 		}
 
-		peerAcc.SetTempRating(newRating)
+		if epoch >= vs.ratingEnableEpoch {
```
Comment (SebastianMarian) [inline process/peer/process.go:877]: So, the other stuff above will be affected. Only the rating will be kept unchanged? peerAcc.IncreaseLeaderSuccessRate(1), peerAcc.SetConsecutiveProposerMisses(0), peerAcc.AddToAccumulatedFees(leaderAccumulatedFees), peerAcc.IncreaseValidatorSuccessRate(1), peerAcc.DecreaseValidatorSuccessRate(1)

File: update/genesis/import.go [hunk 3]
```diff
@@ -61,24 +60,23 @@ func NewStateImport(args ArgsNewStateImport) (*stateImport, error) {
 	if check.IfNil(args.Marshalizer) {
 		return nil, update.ErrNilMarshalizer
 	}
-	if check.IfNil(args.TrieStorageManager) {
+	if args.TrieStorageManagers == nil {
```
Comment (SebastianMarian) [inline update/genesis/import.go:63]: maybe len(args.TrieStorageManagers) == 0

File: genesis/errors.go [hunk 4]
```diff
@@ -115,3 +115,6 @@ var ErrNilNodesSetup = errors.New("nil nodes setup")
 
 // ErrNilDeployProcessor signals that a nil deploy processor has been provided
 var ErrNilDeployProcessor = errors.New("nil deploy processor")
+
+// ErrNilTrieStorageManager signals that a nil trie storage manager has been provided
+var ErrNilTrieStorageManager = errors.New("nil trie storage manager")
```
Comment (SebastianMarian) [inline genesis/errors.go:120]: ErrNilTrieStorageManagers

File: update/genesis/import.go [hunk 5]
```diff
@@ -61,24 +60,23 @@ func NewStateImport(args ArgsNewStateImport) (*stateImport, error) {
...
 		return nil, update.ErrNilTrieStorageManager
```
Comment (SebastianMarian) [inline update/genesis/import.go:64]: ErrNilTrieStorageManagers

File: process/peer/process.go (hunk 2 above)
Comment (sasurobert) [reply process/peer/process.go:877]: yes. those informations are needed to distribute rewards and all others.

Comment (sasurobert) [reply process/peer/process.go:877]: will change to do only the positive changes - no negative. It is better for the well behaving validators.

File: update/genesis/import.go (hunk 3 above)
Comment (sasurobert) [reply update/genesis/import.go:63]: done

File: genesis/errors.go (hunk 4 above)
Comment (sasurobert) [reply genesis/errors.go:120]: done

File: update/genesis/import.go (hunk 5 above)
Comment (sasurobert) [reply update/genesis/import.go:64]: done

File: epochStart/bootstrap/common.go [hunk 6]
```diff
@@ -52,7 +52,7 @@ func checkArguments(args ArgsEpochStartBootstrap) error {
 	if check.IfNil(args.Rater) {
 		return fmt.Errorf("%s: %w", baseErrorMessage, epochStart.ErrNilRater)
 	}
-	if args.TrieStorageManagers == nil {
+	if len(args.TrieStorageManagers) == 0 {
 		return fmt.Errorf("%s: %w", baseErrorMessage, epochStart.ErrNilTrieStorageManager)
```
Comment (SebastianMarian) [inline epochStart/bootstrap/common.go:56]: ErrNilTrieStorageManagers

Comment (sasurobert) [reply epochStart/bootstrap/common.go:56]: done

**Pull request merged**